index.md

```diff
@@ -8,6 +8,573 @@ This is some welcome text.
 
 This is some helpful description of the C0111 error.
 
+
+
+
+### R0201: No self use {#R0201}
+
+If a method (a function in a class) does not make use of the 'self' (or
+first) argument, that means the function is not performing anything that
+is related to the object itself. This means the function could be moved
+outside of the class since none of the code inside makes use of anything
+inside the class it's defined.
+
+```python
+class NoSelfUsage:
+    def __init__(self):
+        self.a = 42
+
+    def no_self(self, num):
+        num = num + 2
+        print(num)
+
+# You would fix it as follows by moving it outside the class:
+def no_self(num):
+	num = num + 2
+    print(num)
+```
+
+
+
+### E0202: Method hidden {#E0202}
+
+If you accidentally mask a method with a field, it can cause other code to
+attempt to invoke what it believes to be a method, which will fail since
+it has become a field instead (a variable). The syntax will error out, and
+this can be particularly dangerous when masking the function when a parent
+class relies on calling the method.
+
+```python
+class Example(object):
+    def field(self, num):
+        return num
+    def __init__(self):
+        self.field = 'Masking the function with this string'
+
+# If you call Example().field(num), it will error out since we masked it
+```
+
+
+
+### E0203: Access to member before definition {#E0203}
+
+Before trying to use a member of a class, it should have been defined at
+some point. If you try to use it before assigning to it, Python cannot
+resolve the value and an error will occur.
+
+```python
+class MyClass:
+	def __init__(self):
+		print(self.a)  # Haven't defined self.a yet, can't use
+		self.a = 5
+```
+
+
+
+### E0211: No method argument {#E0211}
+
+Each method in a class needs to have at least one argument (which is usually
+_self_). Python uses this to call methods, and the first argument is populated
+with the object that is calling the method. This is what allows you to access
+the calling object.
+
+For example, the following two are equivalent:
+
+```python
+class A:
+	def __init__(self):
+		pass
+	def method(self):
+		print('Hi')
+
+# First:
+a = A()
+a.method()
+
+# Second:
+a = A()
+A.method(a)
+```
+
+Therefore, if you do not provide any arguments, then Python does not know
+how to pass the object to the method, and it will error out. To fix this,
+put _self_ (or some variable name) in the parenthesis for the method call.
+
+```python
+class MyClass:
+	def __init__(self):
+		pass
+	def method():
+		print('Missing argument for method definition')
+```
+
+
+
+### E0213: Self as the first argument {#E0213}
+
+The first argument should be the exact word 'self'. This is not an error,
+but it's such a common practice that this is considered an error. The
+following is an example of a good, and bad example:
+
+```python
+class MyClass:
+	def __init__(self):
+		pass
+	
+	def methodA():  # Bad
```

Comment:
This seems more relevant to the error above. Isn't a better example something with a first parameter name that isn't `self`, but something else? (I can't remember if I asked you to fix in this in your pull request in the pyta repo.)